engine-src/Game/LambdaHack/Client/UI/FrameM.hs

```diff
@@ -102,14 +102,14 @@ promptGetKey dm ovs onBlank frontKeyKeys = do
   keyPressed <- anyKeyPressed
   report <- getsSession $ newReport . shistory
   let msgDisturbs = anyInReport disturbsResting report
-  lastPlayOld <- getsSession slastPlay
+  InGameMacro lastPlayOld <- getsSession slastPlay
```

Comment:
I'd rather unwrap the newtype in the `case` below/

Reply:
Funny thing. That was the first thing I did, and then changed it like this just a moment before commit. This is somewhat more optimal or just an aesthetic choice?

Reply:
Exactly the same performance (newtypes induce no overhead). General principle: unwrap (especially invariant-bound) newtypes as late as possible, to document, with types, that the invariant is kept and, generally, if you defined the newtype, you want to use it in favour of the actual type.